chain/ethereum/ethereum.go

```diff
@@ -232,3 +234,71 @@ func BackfillFrom(ctx context.Context, client *ethclient.Client, addresses []com
 		return nil
 	}
 }
+
+//	HistoricalEvents queries past blocks for the events emitted by the given contract addresses.
+//	These events are provided in a channel and ready to be consumed by the caller.
+func HistoricalEvents(ctx context.Context, client *ethclient.Client, addresses []common.Address, fromBlock uint64, toBlock uint64) chan types.Log {
+	if fromBlock == toBlock {
+		return nil
```

Comment:
Return a nil channel can cause unexpexted result for caller who reads from this channel.
You can return a closed channel or return a nil channel with an error instead.